lib/phoenix/live_dashboard/system_info.ex

```diff
@@ -216,25 +216,15 @@ defmodule Phoenix.LiveDashboard.SystemInfo do
 
   defp process_info(pid, prev_reductions) do
     if info = Process.info(pid, @processes_keys) do
-      [{:registered_name, name}, {:initial_call, initial_call} | rest] = info
-
-      name_or_initial_call =
-        if is_atom(name) do
-          inspect(name)
-        else
-          format_initial_call(get_initial_call(pid, initial_call))
-        end
-
       diff = info[:reductions] - (prev_reductions || 0)
 
-      [pid: pid, name_or_initial_call: name_or_initial_call, reductions_diff: diff] ++ rest
-    end
-  end
+      details = ProcessDetails.from(pid)
```

Comment:
Unfortunately you cannot do this, because SystemInfo is loaded into remote nodes, but no other code is. Instead, move `ProcessDetails.from` to this module and call it `to_process_details(...)`.

Reply:
Same for PortDetails.from. In other words, those modules should define only the structs. Perhaps it is even best to define the structs at the top of this file but add a note that the structs are not loaded into the remote node (which is fine because `%ProcessDetails{...}` expands at compile time.